ddtrace/contrib/starlette/patch.py

```diff
@@ -10,16 +12,58 @@
 log = get_logger(__name__)
 
 config._add(
-    "starlette", dict(_default_service="starlette", request_span_name="starlette.request", distributed_tracing=True)
+    "starlette",
+    dict(
+        _default_service="starlette",
+        request_span_name="starlette.request",
+        distributed_tracing=True,
+        aggregate_resources=True,
+    ),
 )
 
 
+def traced_init(wrapped, instance, args, kwargs):
+    global routes
+    routes = kwargs["routes"]
+    mw = kwargs.pop("middleware", [])
+    mw.insert(0, Middleware(TraceMiddleware, integration_config=config.starlette))
+    kwargs.update({"middleware": mw})
+
+    wrapped(*args, **kwargs)
+
+
+def get_resource(scope):
+    path = None
+    global routes
+    for route in routes:
+        match, _ = route.matches(scope)
+        if match == Match.FULL:
+            path = route.path
+            break
+        elif match == Match.PARTIAL and path is None:
+            path = route.path
+    return path
+
+
+def call_decorator(func):
+    def call_wrapper(*args, **kwargs):
+        resource = get_resource(args[1])
```

Comment:
nit: can we pull out `args[1]` into a variable? It would be a lot more readable to read:

```
scope = args[1]
resource = get_resource(scope)
# ....
scope["resource"] = resource
```